### DISCUSSION TITLE
Stucked at installation of web3py.

### DISCUSSION THREAD

**Discussion by naveenburathi** [smartcontractkit/full-blockchain-solidity-course-py#465] [Q&A] [answered]
Stucked at installation of web3py from two days. 
Below is the error I am getting after command : pip install web3

 Running setup.py install for cytoolz ... error
    ERROR: Command errored out with exit status 1:
     command: 'C:\Users\NAVEEN\AppData\Local\Programs\Python\Python310\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\NAVEEN\\AppData\\Local\\Temp\\pip-install-mizt2xvz\\cytoolz_fd7004e1548a43839f1bd819cfc6c8d9\\setup.py'"'"'; __file__='"'"'C:\\Users\\NAVEEN\\AppData\\Local\\Temp\\pip-install-mizt2xvz\\cytoolz_fd7004e1548a43839f1bd819cfc6c8d9\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\NAVEEN\AppData\Local\Temp\pip-record-m90wbxr7\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\Users\NAVEEN\AppData\Local\Programs\Python\Python310\Include\cytoolz'
         cwd: C:\Users\NAVEEN\AppData\Local\Temp\pip-install-mizt2xvz\cytoolz_fd7004e1548a43839f1bd819cfc6c8d9\
    Complete output (53 lines):
    [1/5] Cythonizing cytoolz/utils.pyx
    [2/5] Cythonizing cytoolz/dicttoolz.pyx
    [3/5] Cythonizing cytoolz/functoolz.pyx
    [4/5] Cythonizing cytoolz/itertoolz.pyx
    [5/5] Cythonizing cytoolz/recipes.pyx
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-3.10
    creating build\lib.win-amd64-3.10\cytoolz
    copying cytoolz\compatibility.py -> build\lib.win-amd64-3.10\cytoolz
    copying cytoolz\_signatures.py -> build\lib.win-amd64-3.10\cytoolz
    copying cytoolz\_version.py -> build\lib.win-amd64-3.10\cytoolz
    copying cytoolz\__init__.py -> build\lib.win-amd64-3.10\cytoolz
    creating build\lib.win-amd64-3.10\cytoolz\curried
    copying cytoolz\curried\exceptions.py -> build\lib.win-amd64-3.10\cytoolz\curried
    copying cytoolz\curried\operator.py -> build\lib.win-amd64-3.10\cytoolz\curried
    copying cytoolz\curried\__init__.py -> build\lib.win-amd64-3.10\cytoolz\curried
    copying cytoolz\dicttoolz.pyx -> build\lib.win-amd64-3.10\cytoolz
    copying cytoolz\functoolz.pyx -> build\lib.win-amd64-3.10\cytoolz
    copying cytoolz\itertoolz.pyx -> build\lib.win-amd64-3.10\cytoolz
    copying cytoolz\recipes.pyx -> build\lib.win-amd64-3.10\cytoolz
    copying cytoolz\utils.pyx -> build\lib.win-amd64-3.10\cytoolz
    copying cytoolz\cpython.pxd -> build\lib.win-amd64-3.10\cytoolz
    copying cytoolz\dicttoolz.pxd -> build\lib.win-amd64-3.10\cytoolz
    copying cytoolz\functoolz.pxd -> build\lib.win-amd64-3.10\cytoolz
    copying cytoolz\itertoolz.pxd -> build\lib.win-amd64-3.10\cytoolz
    copying cytoolz\recipes.pxd -> build\lib.win-amd64-3.10\cytoolz
    copying cytoolz\utils.pxd -> build\lib.win-amd64-3.10\cytoolz
    copying cytoolz\__init__.pxd -> build\lib.win-amd64-3.10\cytoolz
    creating build\lib.win-amd64-3.10\cytoolz\tests
    copying cytoolz\tests\dev_skip_test.py -> build\lib.win-amd64-3.10\cytoolz\tests
    copying cytoolz\tests\test_compatibility.py -> build\lib.win-amd64-3.10\cytoolz\tests
    copying cytoolz\tests\test_curried.py -> build\lib.win-amd64-3.10\cytoolz\tests
    copying cytoolz\tests\test_curried_toolzlike.py -> build\lib.win-amd64-3.10\cytoolz\tests
    copying cytoolz\tests\test_dev_skip_test.py -> build\lib.win-amd64-3.10\cytoolz\tests
    copying cytoolz\tests\test_dicttoolz.py -> build\lib.win-amd64-3.10\cytoolz\tests
    copying cytoolz\tests\test_docstrings.py -> build\lib.win-amd64-3.10\cytoolz\tests
    copying cytoolz\tests\test_doctests.py -> build\lib.win-amd64-3.10\cytoolz\tests
    copying cytoolz\tests\test_embedded_sigs.py -> build\lib.win-amd64-3.10\cytoolz\tests
    copying cytoolz\tests\test_functoolz.py -> build\lib.win-amd64-3.10\cytoolz\tests
    copying cytoolz\tests\test_inspect_args.py -> build\lib.win-amd64-3.10\cytoolz\tests
    copying cytoolz\tests\test_itertoolz.py -> build\lib.win-amd64-3.10\cytoolz\tests
    copying cytoolz\tests\test_none_safe.py -> build\lib.win-amd64-3.10\cytoolz\tests
    copying cytoolz\tests\test_recipes.py -> build\lib.win-amd64-3.10\cytoolz\tests
    copying cytoolz\tests\test_serialization.py -> build\lib.win-amd64-3.10\cytoolz\tests
    copying cytoolz\tests\test_signatures.py -> build\lib.win-amd64-3.10\cytoolz\tests
    copying cytoolz\tests\test_tlz.py -> build\lib.win-amd64-3.10\cytoolz\tests
    copying cytoolz\tests\test_utils.py -> build\lib.win-amd64-3.10\cytoolz\tests
    running build_ext
    building 'cytoolz.dicttoolz' extension
    error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'C:\Users\NAVEEN\AppData\Local\Programs\Python\Python310\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\NAVEEN\\AppData\\Local\\Temp\\pip-install-mizt2xvz\\cytoolz_fd7004e1548a43839f1bd819cfc6c8d9\\setup.py'"'"'; __file__='"'"'C:\\Users\\NAVEEN\\AppData\\Local\\Temp\\pip-install-mizt2xvz\\cytoolz_fd7004e1548a43839f1bd819cfc6c8d9\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\NAVEEN\AppData\Local\Temp\pip-record-m90wbxr7\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\Users\NAVEEN\AppData\Local\Programs\Python\Python310\Include\cytoolz' Check the logs for full command output.

**Comment by cromewar** [comment, marked as answer]
Hello @naveenburathi the Cytools error is pretty common for windows users, you have two ways to solve it.
1. Install visual studio community and the python development tools on it.
2. Install WSL (Windows subsystem for linux) and work on a Linux terminal inside windows.

**Reply by naveenburathi** [reply]
thanks it worked

**Reply by katurakes** [reply]
im having the same issue but cant seem to get it to work.

**Reply by cromewar** [reply]
Could share more information @katurakes about what you have tried?
